social_engagement/engagement.py

```diff
@@ -346,11 +413,75 @@ def _get_request(incoming_request, params):
 
 def _get_author_of_comment(parent_id):
     comment = cc.Comment.find(parent_id)
-    if comment and hasattr(comment, 'username'):
-        return comment.username
+    if comment and hasattr(comment, 'user_id'):
+        return comment.user_id
 
 
 def _get_author_of_thread(thread_id):
     thread = cc.Thread.find(thread_id)
-    if thread and hasattr(thread, 'username'):
-        return thread.username
+    if thread and hasattr(thread, 'user_id'):
+        return thread.user_id
+
+
+def _get_thread_details_for_deletion(request, results):
+    """
+    Get details of thread and related users that are required for deletion purposes.
+    """
+    from lms.djangoapps.discussion_api.views import CommentViewSet
```

Comment:
I can't imagine that this is causing a circular import since the LMS surely doesn't import anything from this repo's modules, so we should move this to the top.

Reply:
Ah I see that the other functions are doing it too. I really don't know why they're doing this, though.

Reply:
There is a circular import here:
1 `comment_client.base.views` imports from `social_engagement.engagement`,
2. `discussion_api.api` imports from `comment_client.base.views`,
3. `discussion_api.views` imports from `discussion_api.api`,
4. `social_engagement.engagement` imports from `discussion_api.views`.